File: src/bin/0191-number-of-1-bits.py
# @leetup=info id=191 lang=python3 slug=number-of-1-bits



# @leetup=code
class Solution:
    def hammingWeight(self, n: int) -> int:
        cnt = 0
        while n:
            n = n & (n - 1)
            cnt += 1
        return cnt


# @leetup=code

File: src/bin/0191-number-of-1-bits.rs
// @leetup=info id=191 lang=rust slug=number-of-1-bits

// @leetup=code
impl Solution {
    pub fn hammingWeight(mut n: u32) -> i32 {
        let mut cnt = 0;
        while n != 0 {
            n &= (n - 1);
            cnt += 1;
        }
        cnt
    }
}
// @leetup=code

struct Solution;

fn main() {
    let _ = Solution::hammingWeight;
}
